### PR TITLE
Run tuned in daemon mode.

### PR DESCRIPTION
**Pull request by jmencak**
This PR switches the current one-shot mode of running tuned into
the daemon mode.  This has the following advantages:
  * Settings rollback not only on profile change, but also during
    pod termination.
  * PM QoS support, e.g. /dev/cpu_dma_latency (force_latency)
  * Other functionality needed only available in the daemon mode
    such as dynamic tuning and hotplug.

The disadvantage is a continuous ~13MB additional memory usage by the
tuned daemon.

### CODE REVIEW

**Comment by jmencak** [comment]
/hold
^until https://github.com/openshift/cluster-node-tuning-operator/pull/57 is merged

**Comment by jmencak** [comment]
@sjug PTAL

File: cmd/openshift-tuned.go [hunk 1]
```diff
@@ -201,24 +205,109 @@ func profilesExtract() error {
...
+	}
+	if cmd.Process != nil {
+		glog.V(1).Infof("Sending TERM to PID %d", cmd.Process.Pid)
+		cmd.Process.Signal(syscall.SIGTERM)
+	} else {
+		// This should never happen
+		return fmt.Errorf("Cannot find the tuned process!")
+	}
+	// Wait for tuned process to stop -- this will enable node-level tuning rollback
+	<-tuned_exit
```
Comment (sjug) [inline cmd/openshift-tuned.go:283]: Nice blocking

File: cmd/openshift-tuned.go [hunk 2]
```diff
@@ -201,24 +205,109 @@ func profilesExtract() error {
...
-		_, err = f.WriteString(value)
-		if err != nil {
+		if _, err = f.WriteString(value); err != nil {
 			return fmt.Errorf("Failed to write tuned profile file '%s': %v", profileFile, err)
 		}
 	}
 	return nil
 }
 
+func openshiftTunedPidFileWrite() error {
```
Comment (sjug) [inline cmd/openshift-tuned.go:215]: What are you using this PID file for?

**Review by sjug — COMMENTED** [review]
Looks good

File: cmd/openshift-tuned.go (hunk 2 above)
Comment (jmencak) [reply cmd/openshift-tuned.go:215]: It is used by the pod's `preStop` lifecycle hook to 1) send SIGTERM to openshift-tuned, 2) wait for openshift-tuned to do the actual settings rollback.  I tried to do this without writing an additional PID, but failed.  There might be a more elegant solution without writing the PID, unfortunately I don't know a better way at this point that would work for me.  Another option (without writing a new PID file) might be using the PID written by the tuned daemon itself, but that would also require some additional coding and the current code is probably cleaner.

**Pull request merged**